jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpFields.java

```diff
@@ -1702,9 +1707,9 @@ public void set(HttpField field)
                         }
                         else
                         {
-                            if (last != null && onRemoveField(last))
+                            if (last != null)
                             {
-                                field = onAddField(field);
+                                field = onReplaceField(last, field);
```

Comment:
There is a problem here.   A put method is implemented as an iteration over all fields, setting the first one and then removing the remaining ones.  Unfortunately if there are multiple fields, then that will be seen as "replace", "remove", "remove", which whilst it passes the field value itself, might be a little confusing.   We probably need to do something to make the order: remove, remove, replace.

Reply:
I think I'm caring about this less because I can't see an easy fix.  It is technically correct.

Reply:
Hmm it can be fixed with:
```diff
diff --git a/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpFields.java b/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpFields.java
index 3db9d3a61b..91f61e3eda 100644
--- a/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpFields.java
+++ b/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpFields.java
@@ -1633,6 +1633,37 @@ public interface HttpFields extends Iterable<HttpField>, Supplier<HttpFields>
                 return this;
             }
 
+            @Override
+            public Mutable put(HttpField field)
+            {
+                // rewrite put to ensure that removes are called before replace
+                int put = -1;
+                ListIterator<HttpField> i = listIterator();
+                while (i.hasNext())
+                {
+                    HttpField f = i.next();
+                    if (f.isSameName(field))
+                    {
+                        if (put < 0)
+                            put = i.previousIndex();
+                        else
+                            i.remove();
+                    }
+                }
+
+                if (put < 0)
+                    add(field);
+                else
+                {
+                    for (int n = i.previousIndex() - put; n-- > 0; )
+                        i.previous();
+                    i.previous();
+                    i.set(field);
+                }
+
+                return this;
+            }
+
             @Override
             public Mutable clear()
             {

```